### PR TITLE
refactor: structuring `getColumnInfo` and text alignment logic

### PR DESCRIPTION
**Pull request by haxxmaxx**
We have added things to getColumnInfo incrementally and it has become pretty unstructured and complex. There was some weird casting going on etc. This is mostly to prepare for a bug fix, I just broke this part out.
- renaming align props to more similar names
- moving all alignment logic to functions inside handle-data
- do all dimension specific props in on if in getColumnInfo

### CODE REVIEW

File: src/handle-data.ts [hunk 1]
```diff
@@ -58,6 +59,34 @@ export function getTotalInfo(layout: TableLayout, colIdx: number, pageColIdx: nu
   return '';
 }
 
+/**
+ * Gets the totals alignment for head, totals and body
+ * bodyTextAlign is later used to determine independent alignment for each body cell
+ */
+export function getAlignInfo(
```
Comment (cbt1) [inline src/handle-data.ts:66]: Maybe just me but I find the logic in this function hard to read. Something like this make my head spin a bit less:

```js
export function getAlignInfo(
  textAlign: TextAlign,
  qDimensionType: EngineAPI.DimensionType | undefined,
  isDim: boolean
) {
  if (!textAlign || textAlign.auto) {
    const autoHeadTextAlign = qDimensionType === 'N' || qDimensionType === undefined ? 'right' : 'left';
    const autoTotalsTextAlign = isDim ? 'left' : 'right';

    return {
      headTextAlign: autoHeadTextAlign,
      totalsTextAlign: autoTotalsTextAlign,
      bodyTextAlign: 'auto',
    };
  }

  return {
    headTextAlign: textAlign.align,
    totalsTextAlign: textAlign.align,
    bodyTextAlign: textAlign.align,
  };
}
```

File: src/handle-data.ts [hunk 2]
```diff
@@ -58,6 +59,34 @@ export function getTotalInfo(layout: TableLayout, colIdx: number, pageColIdx: nu
...
+  textAlign: TextAlign,
+  qDimensionType: EngineAPI.DimensionType | undefined,
+  isDim: boolean
+) {
+  const autoHeadTextAlign = qDimensionType === 'N' || qDimensionType === undefined ? 'right' : 'left';
+  const autoTotalsTextAlign = isDim ? 'left' : 'right';
+  const headTextAlign: Align = !textAlign || textAlign.auto ? autoHeadTextAlign : textAlign.align;
+  const totalsTextAlign: Align = !textAlign || textAlign.auto ? autoTotalsTextAlign : textAlign.align;
+  const bodyTextAlign: Align | 'auto' = !textAlign || textAlign.auto ? 'auto' : textAlign.align;
```
Comment (a-m-dev) [inline src/handle-data.ts:75]:
```suggestion
  const bodyTextAlign: Align | 'auto' = !textAlign || textAlign.auto ? textAlign.auto : textAlign.align;
```

maybe?

File: src/handle-data.ts [hunk 3]
```diff
@@ -58,6 +59,34 @@ export function getTotalInfo(layout: TableLayout, colIdx: number, pageColIdx: nu
...
+
+  return { headTextAlign, totalsTextAlign, bodyTextAlign };
+}
+
+/**
+ * Gets the correct text alignment for body cells, based on the text alignment info from the column and cell content
+ */
+export const getBodyCellAlign = (cell: EngineAPI.INxCell, textAlign: Align | 'auto') => {
+  if (textAlign === 'auto') {
+    return ((cell.qNum || cell.qNum === 0) && !Number.isNaN(+cell.qNum) ? 'right' : 'left') as Align;
```
Comment (a-m-dev) [inline src/handle-data.ts:85]:
```suggestion
    return ((cell.qNum || cell.qNum === 0) && Number.isInteger(cell.qNum) ? 'right' : 'left') as Align;
```
is it always integer?

**Review by a-m-dev — APPROVED** [review]
LGTM, just some minor suggestions

File: src/handle-data.ts (hunk 1 above)
Comment (a-m-dev) [reply src/handle-data.ts:66]: may be even dont need  extra variables for 'autoHeadTextAlign' and 'autoTotalsTextAlign'

File: src/handle-data.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (cbt1) [reply src/handle-data.ts:85]: Maybe `qNum` can be an empty string. In which case we want that to align `left`? Assuming `qText` is a string that the we display

Comment (haxxmaxx) [reply src/handle-data.ts:85]: nope can be any number

[the remark this answers is not in the document]
Comment (haxxmaxx) [reply src/handle-data.ts:85]: not sure, @LiKang6688 implemented this, I just moved it

File: src/handle-data.ts [hunk 4]
```diff
@@ -68,6 +97,19 @@ export function getColumnInfo(layout: TableLayout, colIdx: number, pageColIdx: n
...
+  let fieldIndex = 0;
+  let fieldId = '';
+  let isLocked = false;
+  let qDimensionType;
+  if (isDim) {
+    const dimInfo = info as ExtendedNxDimensionInfo;
+    fieldIndex = dimInfo.qGroupPos;
+    fieldId = dimInfo.qGroupFieldDefs[fieldIndex];
+    isLocked = dimInfo.qLocked;
+    ({ qDimensionType } = dimInfo);
```
[the remark this answers is not in the document]
Comment (haxxmaxx) [reply src/handle-data.ts:110]: lint complains about that

File: src/handle-data.ts (hunk 1 above)
Comment (haxxmaxx) [reply src/handle-data.ts:66]: oh wow we check the same thing three times. yes this is way better

File: src/handle-data.ts [hunk 5]
```diff
@@ -58,6 +59,34 @@ export function getTotalInfo(layout: TableLayout, colIdx: number, pageColIdx: nu
...
+ * Gets the totals alignment for head, totals and body
+ * bodyTextAlign is later used to determine independent alignment for each body cell
+ */
+export function getAlignInfo(
+  textAlign: TextAlign,
+  qDimensionType: EngineAPI.DimensionType | undefined,
+  isDim: boolean
+) {
+  const autoHeadTextAlign = qDimensionType === 'N' || qDimensionType === undefined ? 'right' : 'left';
+  const autoTotalsTextAlign = isDim ? 'left' : 'right';
```
[the remark this answers is not in the document]
Comment (haxxmaxx) [reply src/handle-data.ts:72]: going with your suggestion above, no need to name these when we do bengtssons refactor

File: src/handle-data.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (cbt1) [reply src/handle-data.ts:85]: My example code would generate the same result of `qNum` is undefined.

However they would differ if `qNum` equal `"" | null | false`. Not sure how valid those values are. But anyone lets keep it for now unless @LiKang6688 has any comment on it.

[the remark this answers is not in the document]
Comment (haxxmaxx) [reply src/handle-data.ts:85]: agree

File: src/handle-data.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (haxxmaxx) [reply src/handle-data.ts:113]: would be nice to configure so you don't have to destructure when reassigning, but you have to on the first assign (const {something} = obj)

**Review by cbt1 — APPROVED** [review]

**Pull request merged**